x-pack/plugins/fleet/dev_docs/data_model.md

```diff
@@ -37,6 +40,8 @@ All of the code that interacts with this index is currently located in
 [`x-pack/plugins/fleet/server/services/agents/crud.ts`](../server/services/agents/crud.ts) and the schema of these
 documents is maintained by the `FleetServerAgent` TypeScript interface.
 
+- Cleanup model: N/A
```

Comment:
I don't think we remove records from the index. We simply set an `unenrolled_at` value when agents are unenrolled and they're kept around indefinitely.